classes/npm.bbclass

```diff
@@ -19,6 +20,30 @@ ISAR_CROSS_COMPILE = "0"
 SRC_URI = "npm://registry.npmjs.org;name=${NPMPN};version=${PV} \
     ${NPM_SHRINKWRAP}"
 
+BUILDROOT = "${BUILDCHROOT_DIR}/${PP}"
+
+npm_do_mounts() {
+    mkdir -p ${BUILDROOT}
+    sudo mount --bind ${WORKDIR} ${BUILDROOT}
+
+    buildchroot_do_mounts
+}
+
+dpkg_undo_mounts() {
+    i=0
+    while ! sudo umount ${BUILDROOT}; do
+        sleep 0.1
+        if [ `expr $i % 100` -eq 0 ] ; then
+            bbwarn "${BUILDROOT}: Couldn't unmount ($i), retrying..."
+        fi
+        if [ $i -ge 10000 ]; then
+            bbfatal "${BUILDROOT}: Couldn't unmount after timeout"
+        fi
+        i=`expr $i + 1`
+    done
+    sudo rmdir ${BUILDROOT}
+}
```

Comment:
This logic looks generic and should be provided by Isar, not copied into multiple downstream layers.

Reply:
But I assume that Adriaan's mounting cleanup class will resolve that issue upstream. Maybe we should poke him to prioritize that work.